pkg/detectors/endorlabs/endorlabs.go

```diff
@@ -0,0 +1,114 @@
+package endorlabs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyAndSecretPat = regexp.MustCompile(detectors.PrefixRegex([]string{"endor"}) + `\b(endr\+[a-zA-Z0-9]+)\b`)
```

Comment:
question/suggestion: While creating test keys and secrets, I noticed that all were 16 characters long. I'm not sure if this is consistent across all subscription models, as I used the free trial.

If you also observe the 16-character limit after the prefix, we can refine the regex. For example: `\b(endr\+[a-zA-Z0-9-]{16})\b`

Additionally, we should include `-` in the character class since some of the keys I created had hyphens.

Reply:
Yes, it is. I wanted to keep it generic to begin with; maybe we can look at it in the future if things don't seem right.